src/Symfony/Component/HttpFoundation/AcceptHeader.php

```diff
@@ -147,4 +180,145 @@ private function sort(): void
             $this->sorted = true;
         }
     }
+
+    /**
+     * Sorts candidates by specificity, quality, parameter count, and original order.
+     *
+     * @param AcceptHeaderItem[] $candidates
+     */
+    private function sortCandidates(array &$candidates, AcceptHeaderItem $queryItem): void
+    {
+        usort($candidates, function (AcceptHeaderItem $a, AcceptHeaderItem $b) use ($queryItem) {
+            $specificityA = $this->getSpecificity($a, $queryItem);
+            $specificityB = $this->getSpecificity($b, $queryItem);
+            if ($specificityA !== $specificityB) {
+                return $specificityB <=> $specificityA;
+            }
+            if ($a->getQuality() !== $b->getQuality()) {
+                return $b->getQuality() <=> $a->getQuality();
+            }
+            $parameterCountA = \count($this->getMediaParams($a));
+            $parameterCountB = \count($this->getMediaParams($b));
+            if ($parameterCountA !== $parameterCountB) {
+                return $parameterCountB <=> $parameterCountA;
+            }
+
+            return $a->getIndex() <=> $b->getIndex();
+        });
+    }
+
+    /**
+     * Builds a canonical string representation of attributes.
+     *
+     * @param array<string, mixed> $attributes
+     */
+    private function buildCanonicalParams(array $attributes): string
+    {
+        $normalizedAttributes = [];
+        foreach ($attributes as $name => $value) {
+            $lowerName = strtolower($name);
+            if ('q' === $lowerName) {
+                continue;
+            }
+            $normalizedAttributes[$lowerName] = $value;
+        }
+        ksort($normalizedAttributes);
+        $parameterParts = [];
+        foreach ($normalizedAttributes as $name => $value) {
+            if (null === $value) {
+                $parameterParts[] = $name;
+            } else {
+                $quotedValue = \is_string($value) && str_contains($value, ' ') ? '"'.addslashes($value).'"' : $value;
+                $parameterParts[] = $name.'='.$quotedValue;
+            }
+        }
+
+        return implode(';', $parameterParts);
+    }
+
+    /**
+     * Checks if a given header item (range) matches a queried item (value).
+     *
+     * @param AcceptHeaderItem $item      The item from the Accept header (e.g., text/*;format=flowed)
+     * @param AcceptHeaderItem $queryItem The item being queried (e.g., text/plain;format=flowed;charset=utf-8)
+     */
+    private function matches(AcceptHeaderItem $item, AcceptHeaderItem $queryItem): bool
```

Comment:
This whole method feels hard to read and understand. You tend to create temporary variables a lot, I made some suggestions above that try to reduce that. It might be worth revisiting this method for such cases as well? 🙂

Reply:
Trying to comply the RFC adds quite some complexity to this class, which is demonstrated particularly in this method...